pkg/agent/route/route.go

```diff
@@ -310,10 +322,120 @@ func (c *Client) RemoveServiceRouting() error {
 	ipRule := netlink.NewRule()
 	ipRule.IifName = c.nodeConfig.GatewayConfig.Name
 	ipRule.Mark = iptables.RtTblSelectorValue
+	ipRule.Mask = iptables.RtTblSelectorValue
 	ipRule.Table = AntreaServiceTableIdx
 	ipRule.Priority = AntreaIPRulePriority
 	if err = netlink.RuleDel(ipRule); err != nil {
 		return fmt.Errorf("ip rule delete: %w", err)
 	}
 	return nil
 }
+
+// resolveDefaultRouteNHMAC resolves the MAC of default route next
+// hop on service route table.
+func (c *client) resolveDefaultRouteNHMAC() (net.HardwareAddr, error) {
+	// This MAC is relevant to L2 mode only.
+	// TODO, for now uses AKS default MAC.
+	return net.ParseMAC("12:34:56:78:9a:bc")
+}
+
+// SetupPolicyOnlyMode configures routing needed by traffic policy-only mode.
+func (c *client) SetupPolicyOnlyMode() error {
+	gwLink := util.GetNetLink(c.nodeConfig.GatewayConfig.Name)
+	_, gwIP, _ := net.ParseCIDR(fmt.Sprintf("%s/32", c.nodeConfig.NodeIPAddr.IP.String()))
+	if err := netlink.AddrReplace(gwLink, &netlink.Addr{IPNet: gwIP}); err != nil {
+		klog.Fatalf("Failed to add address %s to gw %s: %v", gwIP, gwLink.Attrs().Name, err)
```

Comment:
Should we return here?

Reply:
fatal would crash the process.

Reply:
I have concern about using fatal in library and utils code(see GetNetLink) too, thinking these callees shouldn't have assumption that they are critical enough to control the lifecycle of the whole application, otherwise the control would be spread all over pkg code.
It would be more clean if only main control path decides it and having library code just returning error.

Reply:
I would agree we should not use fatal to exit. Let us return error and handle (finally exit) at the upper layer.

Reply:
sure will change.

Reply:
removed klog.fatal with return fmt.Errorf in route.go
remove klog.fatal in net_linux.go